fastlane/lib/fastlane/helper/sh_helper.rb

```diff
@@ -62,14 +70,15 @@ def self.sh_control_output(*command, print_command: true, print_command_output:
           if error_callback
             UI.error(message)
             error_callback.call(result)
-          else
+          elsif !block_given?
             UI.shell_error!(message)
           end
         end
       else
         result << shell_command # only for the tests
       end
 
+      return block_return if block_given?
```

Comment:
Will this still end up reporting the error if it skips that on line 73 if there is a block?